### ISSUE TITLE
How to deal with lacking RDF conformance of OWL?

### ISSUE THREAD

**Issue by tillmo** [tillmo/DOL#225]
I have moved the discussion about OWL lacking RDF conformance to annex B, so see there. The situation is quite unsatisfactory because OWL is **the** language with RDF conformance (except from RDF itself). Can we relax RDF conformance somehow? Or do have just have to fall back to text conformance? (which would be a bit unsatisfactory)

**Comment by clange** [comment]
tl;dr: I think I found a workaround.  Formally, it is against the OWL specification, but in practice it works.

It would be wrong to say that "OWL lacks RDF conformance". The only problem that we have is that the _serialization of OWL ontologies as RDF graphs_ (BTW regardless of which RDF serialization is used to serialize such graphs, e.g. RDF/XML, Turtle, etc.) does not conform with DOL on the RDF level of conformance.  It does not conform because there is a structural element of the language (imports) to which one cannot assign identifiers.

In the XML Schema that implements the specification of the OWL XML serialization (which is something different from OWL serialized as RDF serialized as RDF/XML), it is not explicitly forbidden to add identifiers to imports, e.g. in the following way:

    <Import xml:id="foo">http://example.com/someOtherOntology</Import>

However, I simply made an experiment by adding to an OWL ontology serialized in RDF/Turtle and having the following triple …

    <URIofImportingOntology> owl:imports <URIofImportedOntology> .

the following "annotation":

    <URIofAnnotation> rdf:subject <URIofImportingOntology> ;
                      rdf:predicate owl:imports ;
                      rdf:object <URIofImportedOntology> ;
                      dct:description "Annotation of import" .

Protégé, which can be considered the de facto reference implementation of OWL, didn't complain about this.  When loading this ontology and saving it, all it did was stripping the rdf:* triples and leaving behind

    <URIofAnnotation> dct:description "Annotation of import" .

i.e. an annotation of something that's not part of the ontology but that has some other URI.

I.e. Protégé doesn't enforce the requirement of the OWL-serialization-as-RDF specification that it is an error if something remains in the RDF input that doesn't match a translation-to-OWL rule ("At the end of this parsing process, the RDF graph must be empty").

Thus we could take the liberty and state that a _variant_ of OWL-serialized-as-RDF is DOL-conforming.  The variant is defined like the original specification plus a rule that matches

    X rdf:subject Y ;
      rdf:predicate Z ;
      rdf:object A ;
      ...other triples... .

and either maps this RDF input to an empty OWL output, or maps it to the same OWL input as if the RDF input had been `X ...other triples... .`.  We might want to fix `Z` to `owl:imports` because I _think_ that in OWL 2 that's the only affected language construct.

Additionally we'd say that any ontology in this _variant_ of OWL-serialized-as-RDF can be translated to a W3C-conforming ontology by removing such parts from the RDF encoding of the ontology. 

Do you like this?

BTW, is the term "import" in DOL the same as "importation"?  If so, it would be better to use only one of the two terms.  If not, this should be clarified somewhere in the standard.

**Comment by tillmo** [comment]
thanks, yes, I like this. Could you please include this into annex B?

I have changed "importation" into "import".

**Comment by tillmo** [comment]
@clange, would you be able to provide some wording today? (Today is the hard OMG deadline for the revision of the document)

**Comment by clange** [comment]
I'm trying to get something quick done _right now_. Sorry but I had a day off today and was offline most of the time.

**Comment by tillmo** [comment]
good, many thanks!

**Comment by clange** [comment]
BTW I should say that in yesterday's commit to Annex B I introduced two minor editorial notes (`\CLnote`), at which you may want to have a look; see https://github.com/tillmo/DOL/commit/4e8d382167fdf778aa992bc1f63587fbaae7cac2

**Comment by tillmo** [comment]
yes, have seen it, thanks! About the capitalization I am not sure...

**Comment by clange** [comment]
Done – strictly spoken, OWL/XML is not XML-conforming either, for a similar reason (I was wrong in my initial comment on this issue). I have now discussed this case as well.

Would it be worth adding the following footnote to the very end of B.1.2. subsubsection "General Issues"?

    \footnote{In practice, many OWL applications, including the Protégé editor, can parse OWL ontologies with imports that carry injected identifiers; they simply ignore these identifiers.}
